lib/usbmux/index.js

```diff
@@ -61,14 +61,26 @@ function swap16 (val) {
  * @throws {B.TimeoutError} if connection timeout happened
  * @returns {Promise<NodeJS.Socket>} Connected socket instance
  */
-async function getDefaultSocket (opts = {}) {
-  const {
+async function getDefaultSocket(opts = {}) {
+  let {
     socketPath = DEFAULT_USBMUXD_SOCKET,
     socketPort = DEFAULT_USBMUXD_PORT,
     socketHost = DEFAULT_USBMUXD_HOST,
     timeout = 5000,
   } = opts;
 
+  if (process.env.USBMUXD_SOCKET_ADDRESS) {
+    // "unix:" or "UNIX:" prefix is optional for unix socket paths.
+    const usbmuxdSocketAddress = process.env.USBMUXD_SOCKET_ADDRESS.replace(/^(unix|UNIX):/i, '');
+    let ipPortPair = usbmuxdSocketAddress.split(':');
+    if (ipPortPair.length === 2) {
+      socketHost = ipPortPair[0];
```

Comment:
not sure if it is a good idea to do the override like that. I would only consider loading them from an environment variable if no explicit values are provided to this method

Reply:
Is there a point in having `opts` here, given the function name implies that it's used to fetch the **default** socket and I don't see that `opts` being filled anywhere in the codebase?

Reply:
it's how it always been historically. You could also consider patching the client code that is using this API. Nevertheless, explicit arguments should always have priority over env variables, otherwise it would be a breaking change.

Reply:
Alright, made the change. if given any `opts` field overrides whatever was set before.